plugins/inputs/rabbitmq/rabbitmq.go

```diff
@@ -448,134 +448,110 @@ func gatherOverview(r *RabbitMQ, acc telegraf.Accumulator) {
-	if numberNodes == 0 {
-		return
-	}
-
-	type NodeCheck struct {
-		NodeName    string
-		HealthCheck HealthCheck
-		Memory      *Memory
-	}
-
-	nodeChecksChannel := make(chan NodeCheck, numberNodes)
-
+	var wg sync.WaitGroup
 	for _, node := range nodes {
-		go func(nodeName string, healthChecksChannel chan NodeCheck) {
-			var healthCheck HealthCheck
-			var memoryresponse MemoryResponse
-
-			err := r.requestJSON("/api/healthchecks/node/"+nodeName, &healthCheck)
-			nodeCheck := NodeCheck{
-				NodeName:    nodeName,
-				HealthCheck: healthCheck,
+		wg.Add(1)
+		go func(node *Node) {
+			defer wg.Done()
+
+			tags := map[string]string{"url": r.URL}
+			tags["node"] = node.Name
+
+			fields := map[string]interface{}{
+				"disk_free":                 node.DiskFree,
+				"disk_free_limit":           node.DiskFreeLimit,
+				"disk_free_alarm":           boolToInt(node.DiskFreeAlarm),
+				"fd_total":                  node.FdTotal,
+				"fd_used":                   node.FdUsed,
+				"mem_limit":                 node.MemLimit,
+				"mem_used":                  node.MemUsed,
+				"mem_alarm":                 boolToInt(node.MemAlarm),
+				"proc_total":                node.ProcTotal,
+				"proc_used":                 node.ProcUsed,
+				"run_queue":                 node.RunQueue,
+				"sockets_total":             node.SocketsTotal,
+				"sockets_used":              node.SocketsUsed,
+				"uptime":                    node.Uptime,
+				"mnesia_disk_tx_count":      node.MnesiaDiskTxCount,
+				"mnesia_disk_tx_count_rate": node.MnesiaDiskTxCountDetails.Rate,
+				"mnesia_ram_tx_count":       node.MnesiaRamTxCount,
+				"mnesia_ram_tx_count_rate":  node.MnesiaRamTxCountDetails.Rate,
+				"gc_num":                    node.GcNum,
+				"gc_num_rate":               node.GcNumDetails.Rate,
+				"gc_bytes_reclaimed":        node.GcBytesReclaimed,
+				"gc_bytes_reclaimed_rate":   node.GcBytesReclaimedDetails.Rate,
+				"io_read_avg_time":          node.IoReadAvgTime,
+				"io_read_avg_time_rate":     node.IoReadAvgTimeDetails.Rate,
+				"io_read_bytes":             node.IoReadBytes,
+				"io_read_bytes_rate":        node.IoReadBytesDetails.Rate,
+				"io_write_avg_time":         node.IoWriteAvgTime,
+				"io_write_avg_time_rate":    node.IoWriteAvgTimeDetails.Rate,
+				"io_write_bytes":            node.IoWriteBytes,
+				"io_write_bytes_rate":       node.IoWriteBytesDetails.Rate,
+				"running":                   boolToInt(node.Running),
 			}
+
+			var health HealthCheck
+			err := r.requestJSON("/api/healthchecks/node/"+node.Name, &health)
 			if err != nil {
 				acc.AddError(err)
-				return
 			}
 
-			err = r.requestJSON("/api/nodes/"+nodeName+"/memory", &memoryresponse)
-			nodeCheck.Memory = memoryresponse.Memory
+			if health.Status == "ok" {
+				fields["health_check_status"] = int64(1)
+			} else {
+				fields["health_check_status"] = int64(0)
+			}
+
+			var memory MemoryResponse
+			err = r.requestJSON("/api/nodes/"+node.Name+"/memory", &memory)
```

Comment:
Is it ok to make another request when the health check failed?  Now that we have "defer wg.done()" shouldn't we return early when health check fails?

Reply:
Second request is safe to do, but is somewhat likely to fail in the case of an earlier error from requestJSON.  It is probably best to not create a partial metric, so I'll change this to return early.